wdl/src/main/scala/wdl/WdlWorkflow.scala

```diff
@@ -155,15 +155,21 @@ case class WdlWorkflow(unqualifiedName: String,
     leftOutputs.find(_.unqualifiedName == name)
   }
 
-  lazy val hasEmptyOutputSection = workflowOutputWildcards.isEmpty && children.collect({ case o: WorkflowOutput => o }).isEmpty
+  lazy val noWorkflowOutputs = children.collect { case o: WorkflowOutput => o }.isEmpty
+  lazy val noOutputWildcards = workflowOutputWildcards.isEmpty
+
+  lazy val hasEmptyOutputSection =  noWorkflowOutputs && noOutputWildcards
+
+  lazy val isTopLevelWorkflow: Boolean =
+    namespace.fullyQualifiedName == ""
 
   /**
-   * All outputs for this workflow and their associated types
+   * All outputs for this workflow and their associated types. Only applies to top-level workflows.
    *
    * @return a Map[FullyQualifiedName, WomType] representing the union
    *         of all outputs from all `call`s within this workflow
    */
-  lazy val expandedWildcardOutputs: Seq[WorkflowOutput] = {
+  lazy val expandedWildcardOutputs: Seq[WorkflowOutput] = if(!ancestry.exists(_.isInstanceOf[WdlWorkflow])) Seq.empty else {
```

Comment:
likewise here, the if and else sides are so imbalanced that it'd be a lot easier to follow if the else side was pulled into its own function or something.

Reply:
Also: this should be using `workflow.isTopLevelWorkflow`, not this weird not-quite-right hack